src/Angle.h

```diff
@@ -0,0 +1,105 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_SG_ANGLE_H_
+#define LSST_SG_ANGLE_H_
+
+/// \file
+/// \author Serge Monkewitz
+/// \brief This file declares a class for representing angles.
+
+#include <cmath>
+#include <algorithm>
+#include <iosfwd>
+#include <limits>
+
+#include "Constants.h"
+
+
+namespace lsst {
+namespace sg {
+
+/// `Angle` represents an angle in radians. It provides methods for
+/// angle comparison and arithmetic, as well as unit conversion.
+/// An angle is said to be normalized if it lies in the range [0, 2π).
+class Angle {
+public:
+    static Angle nan() {
+        return Angle(std::numeric_limits<double>::quiet_NaN());
+    }
+    static Angle fromDegrees(double a) { return Angle(a * RAD_PER_DEG); }
+    static Angle fromRadians(double a) { return Angle(a); }
+
+    /// This constructor creates an Angle with a value of zero.
+    Angle() : _rad(0.0) {}
+    /// This constructor creates an Angle with the given value in radians.
+    explicit Angle(double a) : _rad(a) {}
+
+    // Comparison operators
+    bool operator==(Angle const & a) const { return _rad == a._rad; }
+    bool operator!=(Angle const & a) const { return _rad != a._rad; }
+    bool operator<(Angle const & a) const { return _rad < a._rad; }
+    bool operator>(Angle const & a) const { return _rad > a._rad; }
+    bool operator<=(Angle const & a) const { return _rad <= a._rad; }
+    bool operator>=(Angle const & a) const { return _rad >= a._rad; }
+
+    // Arithmetic operators
+    Angle operator-() const { return Angle(-_rad); }
+    Angle operator+(Angle const & a) const { return Angle(_rad + a._rad); }
+    Angle operator-(Angle const & a) const { return Angle(_rad - a._rad); }
+    Angle operator*(double a) const { return Angle(_rad * a); }
+    Angle operator/(double a) const { return Angle(_rad / a); }
+    double operator/(Angle a) const { return _rad / a._rad; }
+
+    Angle & operator+=(Angle const & a) { *this = *this + a; return *this; }
+    Angle & operator-=(Angle const & a) { *this = *this - a; return *this; }
+    Angle & operator*=(double a) { *this = *this * a; return *this; }
+    Angle & operator/=(double a) { *this = *this / a; return *this; }
+
+    /// `degrees` returns the value of this angle in units of degrees.
+    double degrees() const { return _rad * DEG_PER_RAD; }
+    /// `radians` returns the value of this angle in units of radians.
+    double radians() const { return _rad; }
+
+    /// `isNormalized` returns true if this angle lies in the range [0, 2π).
+    bool isNormalized() const { return _rad >= 0.0 && _rad <= 2.0 * PI; }
```

Comment:
It probably makes sense to have a method to normalize the angle as well as test its normalization status.